validate execution type

mpeck opened the pull request:
Adds a validation to execution in bundle config.

resolves https://github.com/operable/cog/issues/432

christophermaier commented on test/spanner/bundle/validator_test.exs:
```diff
@@ -19,6 +19,10 @@ defmodule Spanner.Bundle.ValidatorTest do
     assert validate("no_templates") == :ok
   end
 
+  test "commands can only have 'once' or 'multiple' execution types" do
```
Testing that `"once"` and `"multiple"` are valid would be good, but this test doesn't do that, so it should be renamed.

Can you write tests that accept YAML from a heredoc instead of a separate file? That would allow fixture data to be close to the tests, and would allow you to generate tests automatically like

```elixir
Enum.each(["once", "multiple"], fn(execution) ->
  test "an execution value of `#{execution}` is valid" do
    #... assertion with YAML in a heredoc
  end
end)
```

mpeck commented:
I updated the tests to make it clearer what's going on. I also moved some of the tests around to match the locations of the tested modules in the code.

kevsmith commented:
:+1:
